include/seqan3/alignment/configuration/align_config_method.hpp

```diff
@@ -28,12 +29,68 @@ namespace seqan3::align_cfg
  *
  * \details
  *
- * The alignment algorithm can be categorised in different methods. For example, the
- * \ref seqan3::align_cfg::method_local "local" and the
- * \ref seqan3::align_cfg::method_global "global" alignment are two different methods, while the semi-global alignment
- * is a variation of the global alignment. This differentiation makes it possible to define a subset of configurations
- * that can work with a particular method. Since it is not possible to guess what the desired method for a user is,
- * there is no default and this configuration must always be provided for the alignment algorithm.
+ * There are several methods for sequence alignment. We distinguish between \ref seqan3::align_cfg::method_local "local"
+ * and \ref seqan3::align_cfg::method_global "global" alignments. The *semi-global* alignment is implemented as a
+ * variation of the global alignment.
+ *
+ * **Global Alignment**:\verbatim
+--T--CC-C-AGT--TATGT-CAGGGGACACG-A-GCATGCAGA-GAC
+  |  || |  ||  | | | |||    || | | |  | ||||   |
+AATTGCCGCC-GTCGT-T-TTCAG----CA-GTTATG-T-CAGAT--C\endverbatim
+ * Finding the optimal global alignment of two sequences is solved by the
+ * [Needleman-Wunsch algorithm](https://en.wikipedia.org/wiki/Needleman%E2%80%93Wunsch_algorithm).
+ *
+ * **Local Alignment** (better suited to find conserved segments):
+ * \verbatim
+                  tccCAGTTATGTCAGgggacacgagcatgcagagac
+                     ||||||||||||
+aattgccgccgtcgttttcagCAGTTATGTCAGatc
+\endverbatim
+ * A \ref seqan3::align_cfg::method_local "local" alignment is effectively a global alignment of two partial sequences.
+ * For example when two genes from different species are similar in short conserved regions and dissimilar in the
+ * remaining regions. A global alignment would not find the local matching because it would try to align the entire
+ * sequence. This is solved by the
+ * [Smith-Waterman algorithm](https://en.wikipedia.org/wiki/Smith%E2%80%93Waterman_algorithm).
+ *
+ * **Semi-global Alignment** (e.g. overlapping sequences):
+ * \verbatim
+                  TCCCAGTTATGTCAGgggacacgagcatgcagagac
+                  |||||||||||||||
+aattgccgccgtcgttttTCCCAGTTATGTCAG
+\endverbatim
+ * The semi-global alignment is a specially configured global alignment, namely we do not penalize gaps at the ends of
+ * the alignment. Semi-global alignments are often used in genome assembly applications when trying to find matching
+ * overlaps.
+ *
+ * **Complexity** of both algorithms:
+ *
+ * Since both algorithms are based on dynamic programming, they run in quadratic time and memory **O(nm)** (where `n`
+ * and `m` are the lengths of the respective aligned sequences).
+ *
+ * To reduce the time complexity you can use a \ref seqan3::align_cfg::band_fixed_size "banded" alignment. It reduces
+ * the runtime by a constant although remaining quadratic and limiting the possible solutions slightly.
```

Comment:
```suggestion
 * To reduce the time complexity you can use a \ref seqan3::align_cfg::band_fixed_size "banded" alignment. A fixed band of size `k` can reduce the runtime to **O(kn)**, but be aware that the quality of the sequence alignments will be worse than an optimal sequence alignment. You should use this option when you know that the two sequences are mostly similar.
```

I think we should mention that a band is orthogonal to the alignment method and e.g. works in combination with global, semi-global and local alignment.

Furthermore, we should mention that depending on the configuration that you can decrease the runtime down to `O(nm / w)` where `w` is typical 64. For the edit distance (myers bitvector, https://docs.seqan.de/seqan/3-master-user/group__alignment__configuration.html#ga3070f8ad7b5692e9c798b411109500a3).

I'm wondering if we should direct to https://docs.seqan.de/seqan/3-master-user/group__alignment__pairwise.html as that already has information about the runtime / space complexity :)

We should change that page to give the space / runtime in terms of two sequences: `O(nm)`